src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -138,7 +141,7 @@
         required    []]
     (first (cli! cli-args specs required))))
 
-(defn run-lifecycle-fn!
+(schema/defn ^:always-validate run-lifecycle-fn! :- a/TrapperkeeperAppContext
```

Comment:
Yeah, that's fair, I just annotated it because that's what it was returning, but I think I agree with your take.  Will remove.